device_doctor/pubspec.lock

```diff
@@ -28,14 +28,14 @@ packages:
       name: async
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "2.5.0"
+    version: "2.5.0-nullsafety.2"
```

Comment:
These needed to be downgraded?

Reply:
These were auto updated when I upgraded the sdk version to sdk: ">=2.3.0 <3.0.0" from sdk: ">=2.1.0 <3.0.0"
(2.3.0 is needed to support Set literal)
Do we see any concerns here?

Reply:
I wonder why upping the minimum dart sdk requirement would downgrade these packages (`2.5.0-nullsafety.2` is older than `2.5.0`).  You'd think if anything it would make you upgrade packages.

I don't really have any concerns I just don't get why it would happen.

Reply:
(the resolved sdk went from `>=2.12.0-0.0 <3.0.0` down to `>=2.10.0-78 <2.11.0`)

Reply:
Ah, that's why. Just learned upping the dart sdk version could possibly lowering the real resolved sdk version.